### DISCUSSION TITLE
Error when calling virtual timeSeries, key 180878

### DISCUSSION THREAD

**Discussion by SIMOGUGE** [Volue-Public/energy-mesh-python#362] [Q&A] [answered]
Hello

We are having an issue when calling a virtual time series through the read_time_series code. You could reproduce when trying it against the latest dump (AXTVOLP1) you received from Axpo.
Timeseries_key: 180878

error message:
![image](https://github.com/Volue-Public/energy-mesh-python/assets/90327853/b5ffb9bd-7feb-4ade-899c-cf17c33a23ab)

While playing around within calculator, I figured out which of the linked underlying vts is causing the issue:
![image](https://github.com/Volue-Public/energy-mesh-python/assets/90327853/060b173d-255d-4f47-8a6b-d1e2055abd29)

Thanks for having a look into it.
Best regards
Simon

**Comment by erny-powel** [comment]
Thanks for the report! We'll have a look internally and let you know what we figure out :smile:

Internal ref: https://github.com/Volue/energy-mesh/issues/3924

**Comment by marthaag** [comment]
Hi @erny-powel did you have a chance to look at it and figure out how we can solve the issue with calling this ts?

**Comment by matburnicki** [comment]
Hi @marthaag,

we've investigated and fixed the issue. The fix will be included in the next Mesh release (2.10).

We've found there were some problems with handling empty lines in the surroundings of comments in the calculations. Before receiving the fix you can use a workaround to fix the calculations.

The workaround is to avoid any empty lines (and white-space) around the comments. Example of a calculation expression that triggers error:

```
## = NaN\n!RestrictionExpand:COMP_KEY,2956\n\n
```

To make it work, just remove the empty line after the comment:

```
## = NaN\n!RestrictionExpand:COMP_KEY,2956\n
```

When trying to read the 180878 timeseries data I've identified the problematic expressions are coming with these referenced timeseries: 36621, 36623, 36625, 36627, 49902, 49904, 49906, 49908.

**Comment by SIMOGUGE** [comment]
Hi @matburnicki 

I tried to use your suggested workaround solution to remove the empty lines after comment on any virutal timeSeries belonging to 180878. The suggested TimsKeys are all for Availability time series (RSS) which cannot be edited except in the availability application.

You can see one example of the expressions in those time series in the below screenshot.
![image](https://github.com/Volue-Public/energy-mesh-python/assets/90327853/b188586c-8493-4faa-a537-b99e8205d4b7)


Best regards
Simon

**Comment by SIMOGUGE** [comment]
Hi @matburnicki , any thoughts on my previous comment.
Best regards
Simon

**Comment by torsannes** [comment, marked as answer]
@SIMOGUGE - there are some additional problems with the calculation formulas containing references to the old availability functionality (like !RestrictionExpand:COMP_KEY). These functions are using information that is not part of Mesh, and are therefore not supported in Mesh calculator. We have done some improvements in the next release for the Mesh calculator in order to give an understandable feedback to the user about unsupported functions in Mesh.
Availability information stored in Mesh is accessible through availability time series stored directly on each Mesh object.

**Comment by SIMOGUGE** [comment]
Hi @torsannes ,
thanks for the details here. I believe this emphasizes how important the correct reflection of revisions and complex restrictions is. In a first step, we would have to replace the availability time series in the currently used virtual time series expressions. Correct?
Best regards
Simon

**Reply by torsannes** [reply]
@SIMOGUGE - as long as you don't plan to use the virtual time series from SmG, it should be correct just to replace the availability time series. I will do an update of your environment this week to have the first test of the revisions between Mesh and SmG.

**Comment by SIMOGUGE** [comment]
Many thanks, @Tor ***@***.***>. We are not entirely sure, as it would logically make a lot more sense to recreate vts contents by purely Mesh-based content. Unfortunately, that is not so easy to achieve on objects like e.g. total rotating reserves over all watercourses. We need to be in a mature state to be able to swap vts “SmG based” with “Mesh based” ones. But we are working on it.

Looking forward to hearing from you. We will have 12.6.5 in Production deployed today.

Best regards
Simon
